packages/core/src/http-handler.ts

```diff
@@ -98,14 +110,24 @@ export class HttpHandler {
   protected _bindInvokeMethod(context: Context) {
     context.bind('invokeMethod').toDynamicValue(() => {
       return async (
-        controllerName: string,
-        method: string,
+        route: ResolvedRoute,
         args: OperationArgs,
       ) => {
+        if (isHandlerRoute(route)) {
+          const result = await route.handler(...args);
+          return result;
+        }
+
+        if (!(route.controllerName && route.methodName)) {
+          throw new Error(
+            'Invalid route: either handler or controllerName + methodName' +
+              ' is required.');
```

Comment:
No no, the message string is saying that both `controllerName` and `methodName` properties on the route object are required.